src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.cs

```diff
@@ -32,6 +93,7 @@ internal JsonTypeInfo(JsonConverter converter, JsonSerializerOptions options)
             }
             private protected set
             {
+                CheckMutable();
```

Comment:
This could be an assert since it's an internal setter.

Reply:
true but it's called by public APIs

Reply:
Generally argument validation should only happen in the boundaries of public APIs.